tcmur_cmd_handler.c

```diff
@@ -1829,7 +1893,14 @@ int tcmur_handle_caw(struct tcmu_device *dev, struct tcmulib_cmd *cmd,
 	if (ret)
 		return ret;
 
-	cmd->cmdstate = caw_fn;
+	rcmd_state = calloc(1, sizeof(*rcmd_state));
+	if (!rcmd_state) {
+		tcmu_dev_err(dev, "Failed to calloc memory for tcmur_cmd_state!\n");
+		return TCMU_STS_NO_RESOURCE;
+	}
+
+	rcmd_state->cmdstate = caw_fn;
+	tcmu_cmd_set_private(cmd, rcmd_state);
 
 	return async_handle_cmd(dev, cmd, tcmur_caw_fn);
```

Comment:
If this fails to queue the cmd to the handler we will leak.

You need to do something like

```
ret = async_handle_cmd()
if (ret != TCMU_STS_ASYNC_HANDLED)
      free(rcmd_state)
```